### PR TITLE
Added Archive and retrieve support for Azure and GCP object storage

### LINKED ISSUE #532: Adding archival and restore of objects into Azure and GCP cloud
**Issue/Feature Description:**
Previously SODA multicloud supported only AWS and now it is enhanced to support Azure and GCP.

**Why this issue to fixed / feature is needed(give scenarios or use cases):**
The Dashboard should allow the user to archive an object to Azure and GCP cloud and also enable retrieve.

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by anvithks**
**What type of PR is this?**
 /kind enhancement

**What this PR does / why we need it**:
This PR adds archive and retrieve(rehydrate) support for Azure and GCP cloud backend object storage.
Previously SODA multicloud supported only AWS and now it is enhanced to support Azure and GCP.
This PR allows the user to use the Dashboard and archive an object to Azure and GCP clouds and retrieve from the same.

**Which issue(s) this PR fixes**:
Fixes #532

**Test Report Added?**:
/kind TESTED

**Test Report**:
archive-azure-1
archive-azure-2
archive-azure-3
image
image

archive-gcp-6
restore-azure-4
restore-azure-5

**Special notes for your reviewer**:

### CODE REVIEW

File: src/app/business/block/bucket-detail/bucket-detail.component.ts [hunk 1]
```diff
@@ -451,20 +450,49 @@ export class BucketDetailComponent implements OnInit {
   //Show Restore Form
   showRestoreObject(file){
     this.restoreDisplay = true;
+    if(this.bucketBackendType=='azure-blob'){
```
Comment (kumarashit) [inline src/app/business/block/bucket-detail/bucket-detail.component.ts:453]: Is it possible to use some constant file and read these from there. So no need to remove across
only change that constant file

File: src/app/shared/utils/consts.ts [hunk 2]
```diff
@@ -84,22 +84,42 @@ export const Consts = {
...
+            {
+                label: "Standard",
+                value: "Standard"
+            },
+            {
+                label: "Bulk",
+                value: "Bulk"
+            }
+        ],
+        'azure-blob' : [
```
Comment (kumarashit) [inline src/app/shared/utils/consts.ts:116]: the storage class change to Hot as well Cold too

Comment (anvithks) [reply src/app/shared/utils/consts.ts:116]: Do you mean add both options here as below:
```
'azure-blob' : [
            {
                label: "Hot",
                value: "Hot"
            },
            {
                label: "Cold",
                value: "Cold"
            }
        ]
```

File: src/app/business/block/bucket-detail/bucket-detail.component.ts (hunk 1 above)
Comment (anvithks) [reply src/app/business/block/bucket-detail/bucket-detail.component.ts:453]: I think it should be possible. let me see what I can do.

**Comment by anvithks** [comment]
@kumarashit Updated the form to be dynamic. It is populated based on the backend type.
Added the Cold Storage Class to Azure restore options.
image
image

File: src/app/shared/utils/consts.ts (hunk 2 above)
Comment (kumarashit) [reply src/app/shared/utils/consts.ts:116]: yes

**Review by kumarashit — APPROVED** [review]
LGTM

**Comment by joseph-v** [comment]
LGTM

**Review by NajmudheenCT — APPROVED** [review]
LGTM

**Pull request merged**